Added endpoints.

skipGit opened the pull request:
Added controllers and services that provides todo app api functionality.

skipGit committed:
Added endpoints.

Added controllers and services that provides todo app api functionality.

yevheniia-shemchenko commented on src/todos/todos.controller.ts:
```diff
@@ -1,4 +1,38 @@
+    @Delete('/:id')
+    deleteTodo(@Param('id') id: number) {
+        this.todosService.deleteTodo(id);
+    }
```
You need to keep the code in one style. 
You should delete unnecessary empty rows.
And if you're putting the empty row after opening and before closing the scope of the method, you should follow those strategy everywhere.

skipGit replied:
Ok, ty.

yevheniia-shemchenko commented on src/todos/todos.service.ts:
```diff
@@ -1,4 +1,40 @@
+export class TodosService {
+  constructor(@InjectModel(Todo) private todoRepository: typeof Todo) {}
+
+  async createToDo(dto: CreateToDoDto) {
+    const todo = await this.todoRepository.create(dto);
+    return todo;
+  }
```
You should rename the method, at other methods you're using Todos not ToDo

skipGit replied:
Ok

yevheniia-shemchenko commented on src/todos/todos.service.ts:
```diff
@@ -1,4 +1,40 @@
+  }
+
+
+  async deleteTodo(id: number) {
+
+    const todo = await this.todoRepository.findOne({where: {id}});
+    this.todoRepository.destroy({where: {id}})
```
You should rename method "destroy".
It' s not clear what the method will do.

MarikNez replied:
Disagree, cause it's looks like a basic method of TS.

skipGit replied:
destroy is sequelize.js built-in method.

yevheniia-shemchenko commented on src/todos/todos.service.ts:
```diff
@@ -1,4 +1,40 @@
+
+  async updateTodo(id: number, dto: CreateToDoDto) {
+    
+    const toUpdate = await this.todoRepository.findOne({where: {id}});
```
You need to rename the name of variable toUpdate.
It's not clear which data contains at this variable.

skipGit replied:
Ok, ty

anniemt6 commented:
As for commit message, you should describe title in imperative style like "Add endpoints", also you should remove point in the end of the title.

anniemt6 commented on src/app.module.ts:
```diff
@@ -2,6 +2,7 @@ import { Module } from "@nestjs/common";
 import { TodosModule } from "./todos/todos.module";
 import { ConfigModule } from '@nestjs/config';
+import { Todo } from "./todos/todos.model";
 
```
Differend code style. Somewhere " ", somewhere ' '.

IvanKrt replied:
@skipGit please describe

skipGit replied:
some of the imports i wrote by myself, but other were generated with autocomplete tools of my IDE. In JSlike languages there is no difference between "packagename" and 'packagename' in import section.

anniemt6 commented on src/todos/todos.controller.ts:
```diff
@@ -1,4 +1,38 @@
+    constructor(private todosService: TodosService) {
+
+    }
+
```
Why did you declare the constructor if it is empty?

skipGit replied:
To use service methods in controller, according to nestjs technic "dependency injection"

anniemt6 commented on src/todos/todos.controller.ts:
```diff
@@ -1,4 +1,38 @@
-import { Controller } from '@nestjs/common';
+import { Body, Controller, Post, Get, Param, Delete, Put } from '@nestjs/common';
```
Code style. Too many dertructured params in one row. You should separate them, enter on the next row.

skipGit replied:
Ok

IvanKrt commented on src/todos/dto/create-todo.dto.ts:
```diff
@@ -0,0 +1,10 @@
+
+
+export class CreateToDoDto {
+
```
Bad practice to leave a lot of free lines.

skipGit replied:
I will work on my code style, ty.

IvanKrt commented on src/todos/todos.service.ts:
```diff
@@ -1,4 +1,40 @@
+    return todo;
+  }
+
+  async updateTodo(id: number, dto: CreateToDoDto) {
```
It is not CreateToDoDto if you use it as an update. Inherit from Create and add id there.

skipGit replied:
In this case service method shares the same dto (data transfer object) with creteToDo (because they are same), but i agree it should be better practice to create another dto for this purpose.

skipGit committed:
Fixed code style issues and reworked Patch endpoint

1. Removed unnecessary whitespaces.
2. Rewrite @Patch service and controller in more elegant form without abusing unnecessary dto.
